### PR TITLE
Update `missing_value_replacement` default

### LINKED ISSUE #576: Impute missing values by default
### Problem Description
Right now, any transformer that can impute missing values does nothing by default. This causes issues when reverse transforming the data, as missing values can be modeled or randomized separately.

### Expected behavior

1. By default, the transformers should impute missing values: `BinaryEncoder` should use `'mode'`. All other transformers should use `'mean'`.
2. If a user explicitly passes in `missing_value_replacement=None`, then throw a warning:

For other `BinaryEncoder`:
```
Warning: Setting the 'missing_value_replacement' to 'None' is no longer supported. Imputing with the 'mode' instead.  
```

For other transformers:
```
Warning: Setting the 'missing_value_replacement' to 'None' is no longer supported. Imputing with the 'mean' instead.  
```

### PR DESCRIPTION
**Pull request by fealho**
Resolve #576.

### CODE REVIEW

File: rdt/performance/performance.py [hunk 1]
```diff
@@ -13,24 +13,13 @@
         'missing_value_replacement': -1,
         'model_missing_values': True
     },
-    'DatetimeTransformer': {
-        'missing_value_replacement': 'mean',
+    'UnixTimestampEncoder': {
```
Comment (fealho) [inline rdt/performance/performance.py:16]: Unrelated change: I think we just forgot to remove these when we renamed the transformers.

**Review by pvk-developer — APPROVED** [review]
LGTM!

File: tests/unit/transformers/test_base.py [hunk 2]
```diff
@@ -362,7 +290,7 @@ def test_is_generator(self):
             ``IS_GENERATOR`` is set to ``True``.
 
         Output:
-            - the boolean value stored in ``COMPOSITION_IS_IDENTITY``.
+            - the boolean value stored in ``IS_GENERATOR``.
```
Comment (amontanez24) [inline tests/unit/transformers/test_base.py:293]: good catch

**Review by amontanez24 — APPROVED** [review]
This LGTM :shipit:
